syndicate/core/resources/eventbridge_schedule_resource.py

```diff
@@ -0,0 +1,149 @@
+    See the License for the specific language governing permissions and
+    limitations under the License.
+"""
+from datetime import datetime, timezone
+
+from syndicate.commons.log_helper import get_logger
+from syndicate.core import ClientError
+from syndicate.core.helper import unpack_kwargs
+from syndicate.core.helper import dict_keys_to_capitalized_camel_case
+from syndicate.core.resources.base_resource import BaseResource
+from syndicate.core.resources.helper import build_description_obj, \
+    validate_params, assert_possible_values
+
+_LOG = get_logger('syndicate.core.resources.eventbridge_schedule_resource')
+
+REQUIRED_PARAMS = {'name', 'schedule_expression', 'state', 'description',
+                   'flexible_time_window'}
+
+
+def convert_to_datetime(name, date_str):
+    try:
+        if len(date_str) > 10:
+            return datetime.fromisoformat(date_str)
+        else:
+            return datetime.utcfromtimestamp(int(date_str))
+    except (ValueError, OSError):
+        raise AssertionError('Invalid date format: {0}. Resource: {1}. Should '
+                             'be ISO8601 or timestamp'.format(
+            date_str, name))
+
+
+def prepare_schedule_parameters(meta):
+    name = meta.get('name')
+    validate_params(name, meta, REQUIRED_PARAMS)
+    params = meta.copy()
+
+    # keys inside "Target" parameters should NOT be changed to PascalCase
+    # syndicate user responsible for providing Target's key-values pairs in proper format
+    target = params.pop('target')
+    params = dict_keys_to_capitalized_camel_case(params)
+    params['Target'] = target
+
+    assert_possible_values([params.get('State')],
+                           ['ENABLED', 'DISABLED']) \
+        if 'State' in params else None
+    assert_possible_values([params.get('FlexibleTimeWindow').get('Mode')],
+                           ['OFF', 'FLEXIBLE']) \
+        if 'Mode' in params.get('FlexibleTimeWindow') else None
+
+    if 'StartDate' in params:
+        start_date = convert_to_datetime(name, params.get('StartDate'))
+        if start_date <= datetime.now(timezone.utc):
+            raise ValueError("Start date must be in the future.")
+    if 'EndDate' in params:
+        end_date = convert_to_datetime(name, params.get('EndDate'))
+        if start_date <= datetime.now(timezone.utc):
+            raise ValueError("End date must be in the future.")
+
+    if 'StartDate' in params and 'EndDate' in params:
+        if start_date >= end_date:
+            raise ClientError("Start date must be earlier than end date.")
+
+    return params
+
+
+class EventBridgeScheduleResource(BaseResource):
+
+    def __init__(self, eventbridge_conn):
+        self.connection = eventbridge_conn
+
+    def create_schedule(self, args):
+        return self.create_pool(self._create_schedule_from_meta, args)
+
+    @unpack_kwargs
+    def _create_schedule_from_meta(self, name, meta):
+        _LOG.debug('Creating schedule %s', name)
```

Comment:
Don't mess several types of string formatting in a single place. 
Just use the most frequently used in the project (f-strings)

Reply:
done